pyiceberg/table/__init__.py

```diff
@@ -1179,6 +1182,11 @@ def refs(self) -> Dict[str, SnapshotRef]:
 
     def _do_commit(self, updates: Tuple[TableUpdate, ...], requirements: Tuple[TableRequirement, ...]) -> None:
         response = self.catalog.commit_table(self, requirements, updates)
+
+        # https://github.com/apache/iceberg/blob/f6faa58/core/src/main/java/org/apache/iceberg/CatalogUtil.java#L527
+        # delete old metadata if METADATA_DELETE_AFTER_COMMIT_ENABLED is set to true
+        self.catalog._delete_old_metadata(self.io, self.metadata, response.metadata)
```

Comment:
similar to https://github.com/apache/iceberg/blob/f6faa58dac57e03be6e02a43937ac7c15c770225/core/src/main/java/org/apache/iceberg/CatalogUtil.java#L539-L544

can we add a comment here explaining how `METADATA_PREVIOUS_VERSIONS_MAX` is taken into account? 
https://github.com/apache/iceberg-python/blob/826a00628216993de50f7f0c2111b6c824b02939/pyiceberg/table/update/__init__.py#L576

Reply:
also maybe we want to wrap this in try/catch and throw a warning as to not block the commit process

Reply:
added a try-catch, but the `deleteFiles` already has a warning for actually deleting the files. I could only see an exception arise from `_delete_old_metadata`'s operations.

https://github.com/apache/iceberg-python/blob/dd175aadfdf03df707bed37008f217258a916369/pyiceberg/catalog/__init__.py#L254-L269